### ISSUE TITLE
KeyError: 'date_submit' when using default=func.now()

### ISSUE THREAD

**Issue by kissgyorgy** [sacrud/pyramid_sacrud#119]
I have this simple model
```python
import sqlalchemy as sa

class User(db.Model):
    name = sa.Column(sa.String(40), nullable=False)
    email = sa.Column(sa.String(100), nullable=False)
    password = sa.Column(db.BcryptPassword)
    joined = sa.Column(sa.DateTime, default=sa.func.now())

    class Meta:
        json_fields = ['name', 'joined']

    def __repr__(self):
        return '<{}: {}>'.format(self.__class__.__name__, self.name)
```

When I try to add a user and leave the joined field empty, I got this traceback:
```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/waitress-0.8.10-py2.7.egg/waitress/channel.py", line 336, in service
    task.service()
  File "/usr/local/lib/python2.7/site-packages/waitress-0.8.10-py2.7.egg/waitress/task.py", line 169, in service
    self.execute()
  File "/usr/local/lib/python2.7/site-packages/waitress-0.8.10-py2.7.egg/waitress/task.py", line 388, in execute
    app_iter = self.channel.server.application(env, start_response)
  File "/usr/local/lib/python2.7/site-packages/whitenoise/base.py", line 119, in __call__
    return self.application(environ, start_response)
  File "/usr/local/lib/python2.7/site-packages/pyramid/router.py", line 223, in __call__
    response = self.invoke_subrequest(request, use_tweens=True)
  File "/usr/local/lib/python2.7/site-packages/pyramid/router.py", line 198, in invoke_subrequest
    response = handle_request(request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/tweens.py", line 20, in excview_tween
    response = handler(request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/router.py", line 145, in handle_request
    view_name
  File "/usr/local/lib/python2.7/site-packages/pyramid/view.py", line 541, in _call_view
    response = view_callable(context, request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/config/views.py", line 602, in __call__
    return view(context, request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/config/views.py", line 328, in attr_view
    return view(context, request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/config/views.py", line 304, in predicate_wrapper
    return view(context, request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/config/views.py", line 385, in viewresult_to_response
    result = view(context, request)
  File "/usr/local/lib/python2.7/site-packages/pyramid/config/views.py", line 497, in _class_view
    response = getattr(inst, attr)()
  File "/usr/local/lib/python2.7/site-packages/ps_alchemy/views.py", line 133, in edit_form_post_view
    deserialized = form.validate_pstruct(pstruct).values()
  File "/usr/local/lib/python2.7/site-packages/deform/field.py", line 661, in validate_pstruct
    cstruct = self.deserialize(pstruct)
  File "/usr/local/lib/python2.7/site-packages/deform/field.py", line 512, in deserialize
    return self.widget.deserialize(self, pstruct)
  File "/usr/local/lib/python2.7/site-packages/deform/widget.py", line 1211, in deserialize
    result[name] = subfield.deserialize(subval)
  File "/usr/local/lib/python2.7/site-packages/deform/field.py", line 512, in deserialize
    return self.widget.deserialize(self, pstruct)
  File "/usr/local/lib/python2.7/site-packages/deform/widget.py", line 1211, in deserialize
    result[name] = subfield.deserialize(subval)
  File "/usr/local/lib/python2.7/site-packages/deform/field.py", line 512, in deserialize
    return self.widget.deserialize(self, pstruct)
  File "/usr/local/lib/python2.7/site-packages/deform/widget.py", line 582, in deserialize
    date_submit = pstruct['date_submit'].strip()
KeyError: 'date_submit'
```

Even if I fill the joined fields.

**Comment by jansselt** [comment]
I submitted the following pull request in the deform project.  We were being held up by the same problem.  Not sure if they'll accept it or not.  But it shows one way around this.

Pylons/deform#298

**Comment by zachzoeller** [comment]
That was the fix I used in my local. Just checked to see if
time/date_submit were none. If not none use them else don't use them.

I think think there are other issues because the front end code acts like
there should be a datetime picker but one isn't rendered in the html.

**Comment by uralbash** [comment]
I added ``pickadate`` js in ``create.jinja2`` template, see https://github.com/sacrud/ps_alchemy/tree/bug-datefield
Could you check my changes in your projects and if all is well then I'll make a new release.

**Comment by uralbash** [comment]
Fixed in ps_alchemy==0.0.6
